src/context-parser-handlebars.js

```diff
@@ -404,17 +431,83 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
 
                 output += t.output;
 
+            // TODO: we support basic partial only, need to enhance it.
+            // http://handlebarsjs.com/partials.html
+            } else if (node.type === handlebarsUtils.PARTIAL_EXPRESSION && 
+                (re = handlebarsUtils.isValidExpression(node.content, 0, node.type)) && 
+                (partialName = re.tag)) {
+
+                ++this._config._partialDepth;
+                if (this._config._partialDepth >= this._config._maxPartialDepth) {
+                    msg = "[ERROR] SecureHandlebars: We detect the partial's depth is more than " + this._config._maxPartialDepth + ", it may cause infinite look up of partial!";
```

Comment:
expect a message like below.
The partial inclusion chain (a > b > c > d) has exceeded the maximum number of depths allowed (maxPartialDepth: 10).